frontend/src/service/Api.service.ts

```diff
@@ -0,0 +1,92 @@
+import { AuthService } from './Auth.service';
+import { Err, ErrorService } from './Error.service';
+import { EnvService } from './Env.service';
+import { GraphQLService, IGraphQLError } from './GraphQL.service';
+import { Url } from '../entity/Url';
+
+export interface IApiService {
+  invokeGetUserShortLinksApi(offset: number, pageSize: number): Promise<Url[]>;
+}
+
+interface IGraphQLApiQuery {
+  authQuery: IGraphQLApiAuthQuery;
+}
+
+interface IGraphQLApiAuthQuery {
+  URLs?: IGraphQLApiURL[];
+}
+
+interface IGraphQLApiURL {
+  alias: string;
+  originalURL: string;
+}
+
+const GET_USER_SHORT_LINKS_QUERY = `
+  query params($authToken: String!) {
+    authQuery(authToken: $authToken) {
+      URLs {
+        alias
+        originalURL
+      }
+    }
+  }
+`;
+
+export class GraphQLApiService implements IApiService {
+  private graphQLBaseURL: string;
+
+  constructor(
+    private authService: AuthService,
+    private errorService: ErrorService,
+    private envService: EnvService,
+    private graphQLService: GraphQLService
+  ) {
+    this.graphQLBaseURL = `${this.envService.getVal(
+      'GRAPHQL_API_BASE_URL'
+    )}/graphql`;
+  }
+
+  invokeGetUserShortLinksApi(offset: number, pageSize: number): Promise<Url[]> {
+    return new Promise((resolve, reject) => {
+      this.graphQLService
+        .query<IGraphQLApiQuery>(this.graphQLBaseURL, {
+          query: GET_USER_SHORT_LINKS_QUERY,
+          variables: { authToken: this.authService.getAuthToken() }
+        })
+        .then((res: IGraphQLApiQuery) => {
+          const { URLs } = res.authQuery;
+          resolve(URLs!.map(this.getUrlFromGraphQLApiUrl));
+        })
+        .catch(err => {
+          const errCodes = this.getGraphQLQueryErrorCodes(err);
+          reject(errCodes[0]);
+        });
+      return;
+    });
+  }
+
+  private getUrlFromGraphQLApiUrl(url: IGraphQLApiURL): Url {
+    return {
+      originalUrl: url.originalURL,
+      alias: url.alias
+    };
+  }
+
+  private getGraphQLQueryErrorCodes(errors: any): string[] {
+    const { networkError, graphQLErrors } = errors;
+    if (networkError) {
+      return [Err.NetworkError];
+    }
+    if (!graphQLErrors || graphQLErrors.length === 0) {
+      return [Err.Unknown];
+    }
+    return graphQLErrors.map(this.gqlErrorToCode);
+  }
+
+  private gqlErrorToCode(graphQLError: IGraphQLError): string {
+    if (graphQLError.message === 'token expired') {
```

Comment:
We should compare code instead of raw message. Message can be easily changed. Code won't. https://github.com/short-d/short/blob/master/frontend/src/service/GraphQL.service.ts#L18

Reply:
Yeah, the issue being, query doesn't return extension code. This is the error response template for graphQL queries.

<img src="https://user-images.githubusercontent.com/18581705/79834725-35a7a380-83cb-11ea-9394-97c868b18c60.png" width="300px" />

Reply:
Interesting. It works for alias already exist error.

Reply:
@rohithbalaji123 Let's return https://github.com/short-d/short/blob/70f3eef11f3fe5a0f27546f970b02ef7019c69c6/backend/app/adapter/graphql/resolver/error.go#L13 at https://github.com/short-d/short/blob/70f3eef11f3fe5a0f27546f970b02ef7019c69c6/backend/app/adapter/graphql/resolver/authquery.go#L52

Reply:
We probably need a switch case here.

Reply:
Shall i create a PR for this part @byliuyang?

Reply:
Why a switch case? what other error types are possible here?

Reply:
@kapillamba4 For now we are just handling unauthenticated error. We may also handle other types of errors separately.

Reply:
@kapillamba4 If you want to open an PR, then it's great!

Reply:
Nice @kapillamba4, thanks for the fix. That saved a lot of time.

Reply:
@rohithbalaji123 He fixed a different package. It's going to take 10 mins to fix.